controllers/Backend.php

```diff
@@ -1,104 +1,344 @@
+            }
+            else {
+                $errors = [];
+            }
+            
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/settings";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        $myView = new View('settings', 'backend');
+        $myView->render(array('modos' => $modos, 'errors' => $errors));
     }
 
     public function list()
     {
-        $model_article = new Article();
-        $model_user = new User();
-
-        $page="backend/list";
-        $topbar="backend/topbar";
-        require "views/layout.php";
-    }
+        $manager_article = new ArticleManager();
+        $posts = $manager_article->get_posts();
 
-    public function logout()
-    {
-        $model_user = new User();
+        if(!isset($_SESSION['admin'])){
+            header("location: /articles-a-la-une");
+        }
         
-        $page="backend/logout";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        $myView = new View('list', 'backend');
+        $myView->render(array('posts' => $posts));
     }
 
     public function password()
     {
-        $model_user = new User();
+        $manager_user = new UserManager();
+        $errors = $manager_user->password_verification();
+   
+         if(filter_has_var(INPUT_POST, 'submit')){
+            $password = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password')));
+            $password_again = filter_var(htmlspecialchars(filter_input(INPUT_POST, 'password_again')));
+
+            $errors = [];
+            
+            if(empty($password) || empty($password_again)){
+                $errors['empty'] = "Tous les champs n'ont pas été remplis";
+            }
+
+            if($password != $password_again){
+                $errors['different'] = "Les mots de passe sont différents";
+            }
+
+            if (preg_match('#^(?=.*[a-z])(?=.*[A-Z]).{8,}$#', $password)) {
+            }
+            else{
+                $errors['non conforme'] = 'Votre mot de passe doit contenir des minuscules et des majuscules et posséder une longueur de 8 caractères au minimum';
+            }	
+                    
+            if(empty($errors)){
+                Usermanager::update_password($password);
+                header("Location:/modification-du-mot-de-passe");
+            }
+        }else
+            $errors =[];
         
-        $page="backend/password";
-        $topbar="backend/topbar";
-        require "views/layout.php";
+        if($manager_user->hasnt_password() == 0){
+        header("Location:/dashboard");
+    }
+        $myView = new View('password', 'backend');
+        $myView->render(array('errors' => $errors));
     }
 
     public function postback()
     {
-        $model_article = new Article();
-        $model_user = new User();
+        if(isset($_GET['id']))
+        {
+        $id = $_GET['id'];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Detected usage of a non-sanitized input variable: $_GET['id']](https://app.codacy.com/manual/JeromeLacquemant/projet-5/pullRequest?prid=5379676)